### DISCUSSION TITLE
Create call graph with tree-sitter

### DISCUSSION THREAD

**Discussion by shaicantor** [tree-sitter/tree-sitter#1605] [Q&A] [answered]
Hi there,

Is there an option to create a call graph with tree-sitter for search based navigation?
Like the article suggests - https://docs.github.com/en/repositories/working-with-files/using-files/navigating-code-on-github#about-navigating-code-on-github

If so, can you please provide some guidelines or guide on how to do that?
Thanks

**Comment by dcreager** [comment, marked as answer]
"Search-based" code navigation doesn't using call graphs, it just looks for textual occurrences of the same symbol name somewhere in the repository.  This is powered by tree-sitter's [tagging functionality](https://github.com/tree-sitter/tree-sitter/tree/master/tags).

"Precise" code navigation is powered by a new framework called stack graphs, which does provide a way to extract call graph information using tree-sitter.  I have a [blog post](https://github.blog/2021-12-09-introducing-stack-graphs/) on the GitHub engineering blog, and an accompanying [talk](https://dcreager.net/talks/2021-strange-loop/) from the Strange Loop conference, that go into more detail about how they work.  Stack graphs are live in production at GitHub for Python, though we're still in the process of migrating over all of the necessary tooling to the tree-sitter open-source repositories.  You can follow the [tree-sitter-graph](https://github.com/tree-sitter/tree-sitter-graph) and [stack-graphs](https://github.com/github/stack-graphs) repos to track the progress.
